table-project/src/app/Providers.tsx

```diff
@@ -0,0 +1,18 @@
+// Providers.jsx
+// This wraps the entire app in root layout.jsx
+"use client";
+
+import { useState } from "react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+
+export default function ReactQueryProvider({
+    children 
+}: {
+    children: React.ReactNode
+}) {
+  const [queryClient] = useState(() => new QueryClient());
+
+  return (
+    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+  );
```

Comment:
This part is pretty shaky to me, but it's more straightforward to demonstrate things if I just wrap the whole root layout.